spot_rl_experiments/spot_rl/envs/base_env.py

```diff
@@ -735,31 +735,43 @@ def get_place_sensor(self):
         # the robot is at the place receptacle
         gripper_T_base = self.get_in_gripper_tf()
         base_T_gripper = gripper_T_base.inverted()
-        base_frame_place_target = self.get_base_frame_place_target()
+        base_frame_place_target = self.get_base_frame_place_target_spot()
         hab_place_target = self.spot2habitat_translation(base_frame_place_target)
         gripper_pos = base_T_gripper.transform_point(hab_place_target)
 
         return gripper_pos
 
-    def get_base_frame_place_target(self):
+    def get_base_frame_place_target_spot(self):
         if self.place_target_is_local:
             base_frame_place_target = self.place_target
         else:
             base_frame_place_target = self.get_target_in_base_frame(self.place_target)
         return base_frame_place_target
 
-    def get_place_distance(self):
-        gripper_T_base = self.get_in_gripper_tf()
-        base_frame_gripper_pos = np.array(gripper_T_base.translation)
-        base_frame_place_target = self.get_base_frame_place_target()
-        hab_place_target = self.spot2habitat_translation(base_frame_place_target)
-        hab_place_target = np.array(hab_place_target)
-        place_dist = np.linalg.norm(hab_place_target - base_frame_gripper_pos)
-        xy_dist = np.linalg.norm(
-            hab_place_target[[0, 2]] - base_frame_gripper_pos[[0, 2]]
+    def get_base_frame_place_target_hab(self):
+        """
+        Returns the current place target in the base frame as (x,z,-y) as per Habitat convention
+        """
+        base_frame_place_target_spot = self.get_base_frame_place_target_spot()
+        base_frame_place_target_hab = np.array(
+            self.spot2habitat_translation(base_frame_place_target_spot)
         )
-        z_dist = abs(hab_place_target[1] - base_frame_gripper_pos[1])
-        return place_dist, xy_dist, z_dist
+        return base_frame_place_target_hab
+
+    # THIS METHOD IS DEPRECATED!!
+    # def get_place_distance(self):
+    #     gripper_T_base = self.get_in_gripper_tf()
+    #     base_frame_gripper_pos = np.array(gripper_T_base.translation)
+    #     base_frame_place_target = self.get_base_frame_place_target_spot()
+    #     hab_place_target = self.spot2habitat_translation(base_frame_place_target)
+    #     hab_place_target = np.array(hab_place_target)
+
+    #     place_dist = np.linalg.norm(hab_place_target - base_frame_gripper_pos)
+    #     xy_dist = np.linalg.norm(
+    #         hab_place_target[[0, 2]] - base_frame_gripper_pos[[0, 2]]
+    #     )
+    #     z_dist = abs(hab_place_target[1] - base_frame_gripper_pos[1])
+    #     return place_dist, xy_dist, z_dist
```

Comment:
@akshararai this method was being used inside .step() methods of place_env, mobile_manipulation_env and lang_env. 

We update all those 3 step functions to use `is_position_within_bounds` for computing the status of if EE of robot reached its goal.

Reply:
Eg lang_env - https://github.com/facebookresearch/spot-sim2real/pull/65/files#diff-2630481e75e183c61cb85e19b4b5d1728739ff458c8a42617be7c3552e32d099R279-R287

Reply:
got it, and the results are equivalent between the two functions?

Reply:
Yes, given the inputs are in same convention.

`get_place_distance` was working in habitat frame. If you pass the coordinates in habitat convetion to `is_position_within_bounds` both are giving exactly same results.

If you however update both points to be in spot convention instead .. the answers might be "slightly" different but that is due to the underlying bugs in spot->habitat convention conversion.